### PR TITLE
[Compute] Update VM SKUs that support accelerated networking

### PR DESCRIPTION
**Pull request by qwordy**
**Description**  

https://github.com/Azure/azure-cli/issues/12858
VM SKUs that support accelerated networking are hard coded in code. Last time when they are updated is in 2018. They are stale now. I change it to fetching from service on the fly so that they are always up to date.

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S168

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 1]
```diff
@@ -714,40 +714,10 @@ def _validate_vm_vmss_accelerated_networking(cli_ctx, namespace):
...
-                      'Standard_L80s_v2', 'Standard_M8ms', 'Standard_M8-4ms', 'Standard_M8-2ms', 'Standard_M16ms',
-                      'Standard_M16-8ms', 'Standard_M16-4ms', 'Standard_M32ms', 'Standard_M32-8ms',
-                      'Standard_M32-16ms', 'Standard_M32ls', 'Standard_M32ts', 'Standard_M64ls', 'Standard_E64i_v3',
-                      'Standard_E64is_v3', 'Standard_E4-2s_v3', 'Standard_E8-4s_v3', 'Standard_E8-2s_v3',
-                      'Standard_E16-4s_v3', 'Standard_E16-8s_v3', 'Standard_E20s_v3', 'Standard_E20_v3']
-        aval_sizes = [x.lower() for x in aval_sizes]
+
+        skus = list_sku_info(cli_ctx, namespace.location)
+        aval_sizes = [x.name.lower() for x in skus if x.resource_type == 'virtualMachines' and
+                      any(c.name == 'AcceleratedNetworkingEnabled' and c.value == 'True' for c in x.capabilities)]
```
Comment (arrownj) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:720]: only list AcceleratedNetworkingEnabled skus ?

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_validators.py:720]: Yes. Then enable accelerated networking for users if the size support this feature. I have some reservations for this design.

**Comment by yungezz** [comment]
hi @qwordy is this PR ready to merge after CI pass?

**Comment by qwordy** [comment]
> 
> 
> hi @qwordy is this PR ready to merge after CI pass?

No. The data is not convincing. I consulted VM people in the issue.

**Comment by yungezz** [comment]
hi @qwordy any update on this PR? thanks

**Comment by qwordy** [comment]
> 
> 
> hi @qwordy any update on this PR? thanks

According to the issue, a service engineer commented 16 days ago. No update yet.

**Comment by qwordy** [comment]
This PR causes more than 100 test failures. It affects all `vm/vmss create` commands.

**Comment by yungezz** [comment]
> This PR causes 100+ test failures in replay mode. It affects all `vm/vmss create` commands. When I rerun them in live mode, still 100+ errors. Most of them are large response errors because list_sku_info returns very large response.

hi @qwordy I remembered that you mentioned add a decorator for this. What's status of it?

**Comment by qwordy** [comment]
> 
> 
> > This PR causes 100+ test failures in replay mode. It affects all `vm/vmss create` commands. When I rerun them in live mode, still 100+ errors. Most of them are large response errors because list_sku_info returns very large response.
> 
> hi @qwordy I remembered that you mentioned add a decorator for this. What's status of it?

The test case author needs to add a large response decorator if the response is large.

**Comment by qwordy** [comment]
This PR fails all vm create tests in replay mode and all of them need to be rerun lively. 
And I don't agree with the existing behavior. Why we make decision for user to enable accelerated networking automatically?
I suggest abandon this PR. The motivation is wrong! We should correct it to the right way, but not improve it in the opposite way.

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 2]
```diff
@@ -706,12 +706,68 @@ def _validate_vm_vmss_accelerated_networking(cli_ctx, namespace):
         size = getattr(namespace, 'size', None) or getattr(namespace, 'vm_sku', None)
         size = size.lower()
 
+        # Deprecated
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:709]: it would be nice to add detail why deprecate

**Comment by yungezz** [comment]
> This PR fails all vm create tests in replay mode and all of them need to be rerun lively.
> And I don't agree with the existing behavior. Why we make decision for user to enable accelerated networking automatically?
> I suggest abandon this PR. The motivation is wrong! We should correct it to the right way, but not improve it in the opposite way.

thanks @qwordy . The original intention is to just by default `on` for user. I don't see problem of default `on`. Will it cause other impact? If the concern is that dynamic `list_sku_infos` too time consuming, can we manual maintain the hardcoded sku list to keep them up-to-date?

**Comment by qwordy** [comment]
> 
> 
> > This PR fails all vm create tests in replay mode and all of them need to be rerun lively.
> > And I don't agree with the existing behavior. Why we make decision for user to enable accelerated networking automatically?
> > I suggest abandon this PR. The motivation is wrong! We should correct it to the right way, but not improve it in the opposite way.
> 
> thanks @qwordy . The original intention is to just by default `on` for user. I don't see problem of default `on`. Will it cause other impact? If the concern is that dynamic `list_sku_infos` too time consuming, can we manual maintain the hardcoded sku list to keep them up-to-date?

The current status of PR does not change default on. It just updates the list. And I comment that how to update the list in the code.

**Comment by yungezz** [comment]
> > > This PR fails all vm create tests in replay mode and all of them need to be rerun lively.
> > > And I don't agree with the existing behavior. Why we make decision for user to enable accelerated networking automatically?
> > > I suggest abandon this PR. The motivation is wrong! We should correct it to the right way, but not improve it in the opposite way.
> > 
> > 
> > thanks @qwordy . The original intention is to just by default `on` for user. I don't see problem of default `on`. Will it cause other impact? If the concern is that dynamic `list_sku_infos` too time consuming, can we manual maintain the hardcoded sku list to keep them up-to-date?
> 
> The current status of PR does not change default on. It just updates the list. And I comment that how to update the list in the code.

yes, got it. above comments are for your previous comments that `should abandon this PR`, and cocnern on existing behavior

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py (hunk 2 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_validators.py:709]: Let me delete these 3 lines.

**Comment by jkewley** [comment]
Point of clarification - the strategy here is to update the hard coded list on a regular basis? Is there some sort of automation or process which will drive a refresh when appropriate, like when VM capabilities are updated??

**Comment by qwordy** [comment]
> 
> 
> Point of clarification - the strategy here is to update the hard coded list on a regular basis? Is there some sort of automation or process which will drive a refresh when appropriate, like when VM capabilities are updated??

I suggest not enable accelerated networking automatically for users. But I keep the original behavior. Currently, no. We are considering supporting it.

**Pull request merged**